lib/src/app_settings/setting.dart

```diff
@@ -1,27 +1,31 @@
 import 'app_settings.dart';
+import 'package:hmi_core/src/core/error/failure.dart';
 
 ///
 /// Holds setting value stored in AppSettings by it name
 /// - value can be returned in int, double or string represantation 
 class Setting {
   final String _name;
   final double _factor;
+  final dynamic Function(Failure err)? _onError;
   ///
   /// - [name] - the name of value stored in AppSettings
   /// - [factor] - returned value int or double will by multiplied by factor
   const Setting(
     String name, {
     double factor = 1.0,
+    dynamic Function(Failure err)? onError,
   }) : 
     _name = name,
-    _factor = factor;
+    _factor = factor,
+    _onError = onError;
   ///
-  /// 
+  /// Returns [Setting] new instance containing a [value]
   const factory Setting.from(dynamic value) = _SettingValue;
```

Comment:
will new instance of `_SettingValue` always throw an error because of `_onError` getter that always throw `UnimplementedError`?